### PR TITLE
Prevents to increment/decrement the value of a readonly or disabled number-field

### PR DESCRIPTION
**Pull request by yeraycaballero**
## 📖 Description
It prevents a number-field value to be incremented or decremented clicking on buttons step-up/step-down, when the number-field is readonly or disabled.
Fixes: #4776

### 🎫 Issues
#4776

## 👩‍💻 Reviewer Notes
It modifies stepDown and stepUp methods to only modify the value if the number-field is not readonly and not disabled
## 📑 Test Plan
Two new tests has been added to number-field.spec.ts.

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Review by nicholasrice — APPROVED** [review]

File: change/@microsoft-fast-foundation-28b13446-5fdd-4c7d-a077-32ec0c284e8a.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fix: it prevents to increment/decrement the value of a readonly or disabled number-field",
```
Comment (awentzel) [inline change/@microsoft-fast-foundation-28b13446-5fdd-4c7d-a077-32ec0c284e8a.json:3]: could say "it prevents incrementing/decrementing the value ..."

**Review by awentzel — APPROVED** [review]

**Review by robarbms — APPROVED** [review]

File: change/@microsoft-fast-foundation-28b13446-5fdd-4c7d-a077-32ec0c284e8a.json (hunk 1 above)
Comment (yeraycaballero) [reply change/@microsoft-fast-foundation-28b13446-5fdd-4c7d-a077-32ec0c284e8a.json:3]: Sure

**Review by EisenbergEffect — APPROVED** [review]

**Comment by EisenbergEffect** [comment]
Pinging @chrisdholt for final review and merge.

**Review by chrisdholt — APPROVED** [review]

**Comment by EisenbergEffect** [comment]
@chrisdholt Ping.

**Comment by chrisdholt** [comment]
> @chrisdholt Ping.

@eisenbergeffect looks like there were multiple PR's for this - one has already been merged.

**Comment by EisenbergEffect** [comment]
@chrisdholt Ok to close this then?

**Comment by chrisdholt** [comment]
Thanks for this @yeraycaballero - closing in favor of #5081 which has gone in. Thanks for helping to drive improvements to FAST!

**Pull request closed without merging**